src/cascadia/QueryExtension/ILLMProvider.idl

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILLMProvider
+    {
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+    }
+
+    interface IResponse
+    {
+        String Message { get; };
+        Boolean IsError { get; };
+    };
+
+    interface IContext
```

Comment:
Does this really need to be an interface (with different implementations?) What would be a second `IContext`? (I can't think of any reason within our codebase that we'd have a second one, or how the `LLMProvider` implementation would deal with different ones)

Reply:
This was designed like this to also set ourselves up for these LLM providers to eventually become extensions/their own separate projects - in which case we just need to show them the interface we expect to communicate to them by (kinda like `IControlSettings`)

Reply:
Sure but like, _we_ (terminal) always implement the _one_ `IContext` implementation, right? There won't ever be `SomeOtherKindOfContext : IContext`, right?

Reply:
_Oh_ I see, so you're suggesting it could just be a struct instead of an interface? Yep that makes sense

Reply:
Chatted over teams, having it be an interface for now makes more sense because we don't want to introduce project dependencies between Terminal and the Extension